### ISSUE TITLE
IndexError: list index out of range

### ISSUE THREAD

**Issue by ashleyuk90** [TheJJ/ceph-balancer#25]
When trying to run the script against my CEPH Cluster either across the whole cluster or a particular pool get the following error:

```
./placementoptimizer.py -v balance --only-pool cephec_ecdata --max-pg-moves 10 | tee /tmp/balance-upmaps 
[2023-03-20 18:47:32,626] gathering cluster state via ceph api...
[2023-03-20 18:47:48,224] running pg balancer
[2023-03-20 18:47:48,224] only considering pools {12}
[2023-03-20 18:47:48,235] current OSD fill rate per crushclasses:
[2023-03-20 18:47:48,236]   hdd: average=75.24%, median=76.04%, crushclass_usage=75.55%
[2023-03-20 18:47:48,237] cluster variance for crushclasses:
[2023-03-20 18:47:48,238]   hdd: 56.797
[2023-03-20 18:47:48,238] min osd.26 53.096%
[2023-03-20 18:47:48,238] max osd.75 90.596%
Traceback (most recent call last):
  File "/root/./placementoptimizer.py", line 2133, in <module>
    pool_pg_count_ideal = pg_mappings.pool_pg_count_ideal(pg_pool, try_pg_move.get_osd_candidates(osd_from))
  File "/root/./placementoptimizer.py", line 885, in get_osd_candidates
    root_name = self.root_names[pg_osd_idx]
IndexError: list index out of range
```

I have a single root in my crush map with all 6 hosts within

**Comment by TheJJ** [comment]
Can you please create and upload your cluster configuration with `./placementoptimizer.py -v gather` and send (a link to) it to `jj at sft dot lol`? Then I can easily reproduce it and try to fix it locally :)

**Comment by TheJJ** [comment]
I've fixed it now, It yields 88.25T new usable space for `cephec_ecdata` in 200 movements :)
